src/_pytest/mark/structures.py

```diff
@@ -37,7 +37,10 @@ def get_empty_parameterset_mark(config, argnames, func):
     if requested_mark in ("", None, "skip"):
         mark = MARK_GEN.skip
     elif requested_mark == "xfail":
-        mark = MARK_GEN.xfail(run=False)
+        if config.getoption("runxfail", default=False):
+            mark = MARK_GEN.skip
```

Comment:
while we should indeed not crash, we shouldn't switch the requested mark to something else

Reply:
to elaborate further - xfail is supposed to eventually be the default

Reply:
Thanks for explaining @RonnyPfannschmidt. Are you OK with `runxfail` taking precedence over the xfail mark though? ( so if I keep the logic, but just move it further up the call stack )

Reply:
Ah sorry, I misunderstood. I'll change it so it properly treats it as xfail in this case